experimental/LieAlgebras/src/WeylGroup.jl

```diff
@@ -7,38 +7,109 @@
 #
 ###############################################################################
 
-struct WeylGroup <: CoxeterGroup
+struct WeylGroup <: GroupsCore.Group
   finite::Bool              # finite indicates whether the Weyl group is finite
-  refl::ZZMatrix            # see positive_roots_and_reflections
+  refl::Matrix{UInt}        # see positive_roots_and_reflections
   root_system::RootSystem   # root_system is the RootSystem from which the Weyl group was constructed
 
-  function WeylGroup(finite::Bool, refl::ZZMatrix, root_system::RootSystem)
+  function WeylGroup(finite::Bool, refl::Matrix{UInt}, root_system::RootSystem)
     return new(finite, refl, root_system)
   end
 end
 
+struct WeylGroupElem <: GroupsCore.GroupElement
+  parent::WeylGroup     # parent group
+  word::Vector{UInt8}   # short revlex normal form of the word
+
+  function WeylGroupElem(W::WeylGroup, word::Vector{<:Integer}; normalize::Bool=true)
+    if !normalize
+      if word isa Vector{UInt8}
+        return new(W, word)
+      else
+        return new(W, UInt8.(word))
+      end
+    end
+
+    @req all(1 <= i <= ngens(W) for i in word) "word contains invalid generators"
+    x = new(W, sizehint!(UInt8[], length(word)))
+    for s in Iterators.reverse(word)
+      lmul!(x, s)
+    end
+
+    return x
+  end
+end
+
+const IteratorNoCopyState = Tuple{WeightLatticeElem, WeylGroupElem}
```

Comment:
Please put next to the iterator definition and rename it `WeylIteratorNoCopyState` or something similar that expresses meaning